### PR TITLE
Part-1 code

### PR DESCRIPTION
**Pull request by ChrisNewmanGDL**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -76,6 +76,101 @@ returning lhs
...
+    }
+    float divide( float lhs, float rhs )
+    {
+        if(rhs == 0.0f)
+        { 
+            std::cout << "\nwarning error, integer division by zero returns 'inf' !"  << std::endl;
+        }
+        
+            return lhs / rhs;
+    }
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation.   Train your eye to spot bad indentation
- [x] revise

File: main.cpp [hunk 2]
```diff
@@ -76,6 +76,101 @@ returning lhs
...
+    double divide( double lhs, double rhs )
+    {
+         if(rhs == 0.0)
+        { 
+            std::cout << "\nwarning error, integer division by zero returns 'inf' !"  << std::endl;
+        }
+        
+            return lhs / rhs;
+    }
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation.   Train your eye to spot bad indentation
- [x] revise

File: main.cpp [hunk 3]
```diff
@@ -76,6 +76,101 @@ returning lhs
...
+        if(rhs == 0.0f)
+        { 
+            std::cout << "\nwarning error, integer division by zero will crash the program!"  << std::endl;
+            std::cout << "returning lhs" << std::endl << lhs << std::endl;
+            return 0;
+        }
+                    
+            return lhs / rhs;
+                         
+    }
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation.   Train your eye to spot bad indentation
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp [hunk 4]
```diff
@@ -76,6 +76,101 @@ returning lhs
...
+    {
+        return lhs - rhs;
+    }
+    int multiply( int lhs, int rhs )
+    {
+        return lhs * rhs;
+    }
+    int divide( int lhs, int rhs )
+    {
+        if(rhs == 0.0f)
```
Comment (matkatmusic) [inline main.cpp]: don't compare `int` with `float`
compare `int` with `int`
- [x] revise

File: main.cpp [hunk 5]
```diff
@@ -76,6 +76,101 @@ returning lhs
...
+        { 
+            std::cout << "\nwarning error, integer division by zero will crash the program!"  << std::endl;
+            std::cout << "returning lhs" << std::endl << lhs << std::endl;
+            return 0;
```
Comment (matkatmusic) [inline main.cpp]: You aren't returning `lhs`, like the output message says.
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 5 above)
Comment (ChrisNewmanGDL) [reply main.cpp]: ah, that was me lacking insight into the task than carelessness, (I carefully printed lhs!). I get what you were asking for now.

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**